18/s2i/bin/run

```diff
@@ -20,11 +20,34 @@ run_node() {
   if [ "$DEV_MODE" == true ]; then
     echo "Launching via nodemon..."
     exec nodemon --inspect="$DEBUG_PORT"
+  elif [ -v NODE_CMD ]; then
+    if [ "$INIT_WRAPPER" == true ]; then
+      echo "launching via init wrapper..."
+      exec ${STI_SCRIPTS_PATH}/init-wrapper $NODE_CMD
+    else
+      echo "Launching via ${NODE_CMD}"
+      echo "exec $NODE_CMD"
+    fi
+  elif [ "$INIT_WRAPPER" == true ]; then
+
+    if [ -f "server.js" ]; then
+      start_file="server.js"
+    elif [ -f "index.js" ]; then
+      start_file="main.js"
```

Comment:
looks like a typo should be index.js ?  Makes me wonder if there is anywhere we could add some testing?

Reply:
Also are these three files documented in the documentation somewhere else? Otherwise we might want to add that to the documentation as well.

Reply:
I also wondered if we could pull what npm start would have used from package.json ?

Reply:
> Also are these three files documented in the documentation somewhere else? Otherwise we might want to add that to the documentation as well.

We only mention the presence of these files when we describe the init-wrapper functionality. I dont think we should mention something more because is specific to the scope of how init-wrapper works - looks for the files.

Reply:
> I also wondered if we could pull what npm start would have used from package.json ?

@mhdawson 
Sure,
by looking at the npm start documentation https://docs.npmjs.com/cli/v9/commands/npm-start?v=true I tried to reproduce the start command and I ended up with below implementation which will replace the current one of this pr https://github.com/pacostas/s2i-nodejs-container/blob/a59f33d729564d7509b56680cfd84a7406f9a5ca/18/s2i/bin/run#L31-L45

```bash
elif [ "$INIT_WRAPPER" == true ]; then

    package_json_start=$(cat package.json | grep "start" | sed -r 's/(.*)"start"([^:]*[^"]*")(.*)",/\3/')
    package_json_main=$(cat package.json | grep "main" | sed -r 's/(.*)"main"([^:]*[^"]*")(.*)",/\3/')

    if [ -n "$package_json_start" ]; then
        start_command=$package_json_start
    elif [ -n $package_json_main ]; then
        start_command="node ."
    elif [ -f "server.js" ]; then
        start_command="node server.js"
    else
        echo "Failed to find file for starting the Node.js application"
        exit 1
    fi

    echo "launching via init wrapper..."
    exec ${STI_SCRIPTS_PATH}/init-wrapper $start_command
else
```

Reply:
> > I also wondered if we could pull what npm start would have used from package.json ?
> 
> Yes we can do that with bash code, working on it.

@mhdawson 
by looking at the npm start documentation https://docs.npmjs.com/cli/v9/commands/npm-start?v=true I tried to reproduce the start command and I ended up with below implementation which will replace the current one of this pr https://github.com/pacostas/s2i-nodejs-container/blob/a59f33d729564d7509b56680cfd84a7406f9a5ca/18/s2i/bin/run#L31-L45

```bash
elif [ "$INIT_WRAPPER" == true ]; then

    package_json_start=$(cat package.json | grep "start" | sed -r 's/(.*)"start"([^:]*[^"]*")(.*)",/\3/')
    package_json_main=$(cat package.json | grep "main" | sed -r 's/(.*)"main"([^:]*[^"]*")(.*)",/\3/')

    if [ -n "$package_json_start" ]; then
        start_command=$package_json_start
    elif [ -n $package_json_main ]; then
        start_command="node ."
    elif [ -f "server.js" ]; then
        start_command="node server.js"
    else
        echo "Failed to find file for starting the Node.js application"
        exit 1
    fi

    echo "launching via init wrapper..."
    exec ${STI_SCRIPTS_PATH}/init-wrapper $start_command
else
```